src/input.rs

```diff
@@ -127,6 +131,87 @@ pub trait ReadBytes {
     }
 }
 
+// # Safety
+//
+//   - `read_into_uninit()` and `read_into_uninit_exact()` do return prefix
+//     subslices of their `buffer`s
+unsafe
+impl<R: io::Read> uninit::ReadIntoUninit for BufferedReader<R>
+{
+    #[inline]
+    fn read_into_uninit<'buf> (
+        self: &'_ mut Self,
+        buffer: &'buf mut [MaybeUninit<u8>],
+    ) -> io::Result<&'buf mut [u8]>
+    {
+        self.read_into_uninit_exact(buffer)
```

Comment:
I wanted to avoid code repetition for a start, later on this could be improved to use `.read_into_uninit_exact`'s implementation, but replacing the EOF error with a return of the initialized subslice.